app/main.py

```diff
@@ -1,3 +1,69 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    # +
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+
+        return Distance(km=self.km + other)
+
+    # +=
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, (int, float)):
+            self.km += other
+            return self
+
+        self.km += other.km
+        return self
+
+    # *
+    def __mul__(self, other: int) -> Distance:
+        return Distance(km=self.km * other)
+
+    # /
+    def __truediv__(self, other: int | float) -> Distance:
+        if other == 0:
+            return Distance(other)
+        result = round(self.km / other, 2)
+        return Distance(km=result)
+
+    def __lt__(self, other: Distance) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km < other
+
+        return self.km < other.km
+
+    def __gt__(self, other: Distance) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km > other
+
+        return self.km > other.km
+
+    def __eq__(self, other: Distance) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km == other
+
+        return self.km == other.km
+
+    def __le__(self, other: Distance) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km <= other
+
+        return self.km <= other.km
+
+    def __ge__(self, other: Distance) -> bool:
+        if isinstance(other, (int, float)):
```

Comment:
The type hint for the `other` parameter should include `int` and `float` as well as `Distance`. Update it to `other: Distance | int | float`.